### ISSUE TITLE
How does inference work? Always getting same predictions

### ISSUE THREAD

**Issue by kuhanw** [ematvey/tensorflow-seq2seq-tutorials#27]
Hi all,

I am currently trying to perfom inference on a task using the example of the third tutorial notebook as my guide. 

The input is a sequence of english language sentences and the output is also an english language phrase or sentence.

My problem is at inference time all my input sentences are returning the same output sentence. Even my training examples. I have to think this is a problem with how I am executing inference because I can clearly see in the training output that the predictions are matching the training examples and the loss decreases to <0.1. 

Here are some relevant snippets of my code:
------------------------------------------------------------
model = Seq2SeqModel(encoder_cell=LSTMCell(250),
                     decoder_cell=LSTMCell(500), 
                     vocab_size=len(vocab_dict),
                     embedding_size=300,
                     attention=True,
                     bidirectional=True,
                     debug=False)

session.run(tf.global_variables_initializer())
saver = tf.train.Saver()    

if train:
    for epoch in range(3001):
        train_model(epoch, session, model,
                       length_from=df_all['n_quote'].min(), length_to=df_all['n_quote'].max(),
                       vocab_lower=2, vocab_upper=len(vocab_dict),
                       batch_size=batch_size,
                       max_batches=batches_in_epoch, verbose=True)

        if epoch % 500 ==0: saver.save(session, 'seq2seq_fixed_labels', global_step=epoch)

else:

    saver.restore(session, './seq2seq_fixed_labels_v4-1000')
    fd = model.make_inference_inputs([encodeSentence('Next we year we will focus on retention'), \
                                      encodeSentence('Cybersecurity incidents could disrupt business operations  result in the loss of critical and confidential information  and adversely impact our reputation and results of operations'),
                                      encodeSentence('Expanding into emerging markets is important for us'), \
                                      encodeSentence('In closing we had a strong start to the year and we expect this performance to continue throughout the balance of')])

    inf_out = session.run(model.decoder_prediction_inference, fd)
    inf_prob_out = session.run(model.decoder_prediction_prob_inference, fd)

    print (inf_out)
    for i, (e_in, dt_pred, dt_pred_prob) in enumerate(zip(fd[model.encoder_inputs].T, 
                             inf_out.T, inf_prob_out.T)):
        #if i > 0: break
        print('    sample {}:'.format(i + 1))
        print('    enc input                > {}'.format([inv_map[i] for i in e_in]))
        print('    dec train predicted      > {}'.format([inv_map[i] for i in dt_pred]))
        print('    dec train predicted prob > {}'.format(dt_pred_prob))

--------------------------------------------------------------------

The predicted output of this is always the sentence string:

u'Goal', u'', u'', u'A', u'highly', u'effective', 'majoreconomic', u'engine', u'creating', 'newpartnerships', u'to', u'build', u'a', 'strongand', u'sustainable', u'future', u'for', u'', u'Florida', u'in', u'the', u'global', u'economy', '<EOS>'

I have checked that the input "fd" is different for every block of text going into model.decoder_prediction_inference. But the output is always the same. Any ideas?

Thank You,

Kuhan

**Comment by kuhanw** [comment]
Turned out to be a bug in my code. At the inference time the size of my vocabulary was larger than at training time. I finally tracked it down by trying to diagnose the error on the trivial copy task problem by training the copy-task problem with a massive dictionary size to emulate the real problem.

**Comment by ghost** [comment]
Hi @kuhanw , I am having same problem. my output is always same for all sentences during inference time. could you please throw more light on how U fixed this? Thanks
